src/SigasiProjectCreator/DotF/DotFfileParser.py

```diff
@@ -64,45 +70,74 @@ def __init__(self, filename):
+                elif bare_option.startswith("+") or bare_option.startswith("-"):
+                    print(f'*.f parse* unknown option (ignored) : {bare_option}')
+                elif parser_expect_dot_f:
+                    parser_expect_dot_f = False
                     # Parse included .f file
-                    subfile = os.path.expandvars(bare_option.split()[1])
+                    subfile = expandvars_plus(bare_option)
                     if not os.path.isabs(subfile):
                         subfile = os.path.join(self.dotfdir, subfile)
                     subparser = DotFfileParser(subfile)
                     self.library_mapping.update(subparser.library_mapping)
                     self.includes |= subparser.includes
                     self.defines.extend(subparser.defines)
-                elif bare_option.startswith("+") or bare_option.startswith("-"):
-                    print("Unknown option (ignored) : " + bare_option)
+                elif parser_expect_library:
+                    # new library name
+                    parser_expect_library = False
+                    newlib = bare_option.split('/')[-1]
                 else:
                     # Design file: add to library mapping
-                    if "*" in bare_option:
-                        expanded_option = glob.glob(rebase_file(bare_option, self.dotfdir), recursive=True)
-                        for f in expanded_option:
-                            self.add_to_library_mapping(f, 'work')
+                    if str(os.path.splitext(bare_option)[1]).lower() in ['.vhd', '.vhdl', '.v', '.sv']:
+                        self.add_to_library_mapping(bare_option, newlib)
                     else:
-                        self.add_to_library_mapping(rebase_file(bare_option, self.dotfdir), 'work')
+                        print(f'*.f parse* skipping {bare_option}')
 
     def add_to_library_mapping(self, file, library):
-        if str(ArgsAndFileParser.get_layout_option()) == 'default':
-            if file in self.library_mapping:
-                file_base, file_ext = os.path.splitext(file)
-                newfile = file_base + '_' + library + file_ext
-                if newfile in self.library_mapping:
-                    print('File already mapped to library: ' + file + ' => ' + library)
-                else:
-                    self.library_mapping[newfile] = library
-                    self.linked_file_mapping[newfile] = file
-            else:
-                self.library_mapping[file] = library
+        # For now, we'll return a dict of abs path => library
+        # TODO Problem (to be re-solved): files may be mapped to more than one library, in which case this
+        #  approach won't work
+        # Layout moves to ConverterHelper
+
+        # We need to expand environment variables here, before expanding wildcards
+        expanded_path = os.path.expandvars(file)
+        if "*" in expanded_path:
```

Comment:
Could the `*` not be escaped here?

Reply:
I don't think that an escape is needed here. We test for a wildcard in and handle it if it's present.